### ISSUE TITLE
[SkyServe] Call back functions for sensitive action

### ISSUE THREAD

**Issue by cblmemo** [skypilot-org/skypilot#2918]
Like managed spot jobs (#2106), we should support customized notification for sensitive notification, e.g., replica failed, scaling options and more.

**Comment by kyuds** [comment]
@cblmemo just wondering if this is still relevant

**Comment by cblmemo** [comment]
> [@cblmemo](https://github.com/cblmemo) just wondering if this is still relevant

Yes, this is still relevant. We will try to get to it once we got enough bandwidth!

**Comment by kyuds** [comment]
oh I asked this more because I wanted to try and handle this issue in the near future!

**Comment by cblmemo** [comment]
> oh I asked this more because I wanted to try and handle this issue in the near future!

Sounds great. We dont have any particular person who is working on this rn so feel free to take a stab!
